salt/states/openvswitch_port.py

```diff
@@ -37,73 +37,79 @@ def present(name, bridge, type=None, id=None, remote=None, dst_port=None, intern
-                                   'new': 'Created GRE tunnel interface {0} with remote ip {1} and key {2} '
-                                          'on bridge {3}.'.format(name, remote, id, bridge),
-                                   }
-                            }
+    if type:
+        comments['comment_invalid_ip'] = 'Remote is not valid ip address.'
+        if type == "vlan":
+            comments['comment_vlan_invalid_id'] = 'VLANs id must be between 0 and 4095.'
+            comments['comment_vlan_invalid_name'] = 'Could not find network interface {0}.'.format(name)
+            comments['comment_vlan_port_exists'] = 'Port {0} with access to VLAN {1} already exists on bridge {2}.'.format(name, id, bridge)
+            comments['comment_vlan_created'] = 'Created port {0} with access to VLAN {1} on bridge {2}.'.format(name, id, bridge)
+            comments['comment_vlan_notcreated'] = 'Unable to create port {0} with access to VLAN {1} on ' \
+                                      'bridge {2}.'.format(name, id, bridge)
+            comments['changes_vlan_created'] = {name: {'old': 'No port named {0} with access to VLAN {1} present on '
+                                                  'bridge {2} present.'.format(name, id, bridge),
+                                           'new': 'Created port {1} with access to VLAN {2} on '
+                                                  'bridge {0}.'.format(bridge, name, id),
+                                           }
+                                    }
 
-    comment_dstport = ' (dst_port' + str(dst_port) + ')' if 0 < dst_port <= 65535 else ''
-    comment_vxlan_invalid_id = 'Id of VXLAN tunnel must be an unsigned 64-bit integer.'
-    comment_vxlan_interface_exists = 'VXLAN tunnel interface {0} with rempte ip {1} and key {2} ' \
-                                   'already exists on bridge {3}{4}.'.format(name, remote, id, bridge, comment_dstport)
-    comment_vxlan_created = 'Created VXLAN tunnel interface {0} with remote ip {1}  and key {2} ' \
-                          'on bridge {3}{4}.'.format(name, remote, id, bridge, comment_dstport)
-    comment_vxlan_notcreated = 'Unable to create VXLAN tunnel interface {0} with remote ip {1} and key {2} ' \
-                             'on bridge {3}{4}.'.format(name, remote, id, bridge, comment_dstport)
-    changes_vxlan_created = {name: {'old': 'No VXLAN tunnel interface {0} with remote ip {1} and key {2} '
-                                         'on bridge {3}{4} present.'.format(name, remote, id, bridge, comment_dstport),
-                                   'new': 'Created VXLAN tunnel interface {0} with remote ip {1} and key {2} '
-                                          'on bridge {3}{4}.'.format(name, remote, id, bridge, comment_dstport),
-                                   }
-                            }
-    comment_invalid_ip = 'Remote is not valid ip address.'
+        elif type == "gre":
```

Comment:
This shadows the Python built-in of `type`. I don't think this is doing what you intend here.